src/Draco.Compiler.Tests/Syntax/ParserTests.cs

```diff
@@ -114,4 +204,1269 @@ public void TestEmptyFuncWithoutClosingCurly()
+                                    this.T(TokenType.LiteralInteger);
+                                }
+                            }
+                            this.T(TokenType.Semicolon);
+                        }
+                    }
+                }
+                this.T(TokenType.CurlyClose);
+                this.N<Expr.ElseClause>();
+                {
+                    this.T(TokenType.KeywordElse);
+                    this.N<Expr.UnitStmt>();
+                    this.N<Stmt.Expr>();
+                    this.N<Expr.Block>();
+                    this.T(TokenType.CurlyOpen);
+                    this.N<BlockContents>();
+                    {
+                        this.N<Stmt.Decl>();
+                        {
+                            this.N<Decl.Variable>();
+                            {
+                                this.T(TokenType.KeywordVal);
+                                this.T(TokenType.Identifier);
+                                this.N<ValueInitializer>();
+                                {
+                                    this.T(TokenType.Assign);
+                                    this.N<Expr.Literal>();
+                                    {
+                                        this.T(TokenType.LiteralCharacter);
+                                    }
+                                }
+                                this.T(TokenType.Semicolon);
+                            }
+                        }
+                    }
+                    this.T(TokenType.CurlyClose);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("""
+            if(5 > 0){
+                val x = 5;
+            }
+            else{
+                val y = 'c';
+            }
+            """, IfElse);
+    }
+
+    [Fact]
+    public void TestElseStatementsMissingIf()
+    {
+        void OnlyElse()
+        {
+            this.N<Stmt.Unexpected>();
+            {
+                this.T(TokenType.KeywordElse);
+                this.N<Expr.Block>();
+                this.T(TokenType.CurlyOpen);
+                this.N<BlockContents>();
+                {
+                    this.N<Stmt.Decl>();
+                    {
+                        this.N<Decl.Variable>();
+                        {
+                            this.T(TokenType.KeywordVal);
+                            this.T(TokenType.Identifier);
+                            this.N<ValueInitializer>();
+                            {
+                                this.T(TokenType.Assign);
+                                this.N<Expr.Literal>();
+                                {
+                                    this.T(TokenType.LiteralInteger);
+                                }
+                            }
+                            this.T(TokenType.Semicolon);
+                        }
+                    }
+                }
+                this.T(TokenType.CurlyClose);
```

Comment:
This does not look correct, I don't think this belongs to the unexpected node.

Reply:
We could definitely _make_ it belong to the else node tho, but currently the test is a bit misleading.